src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -351,12 +351,15 @@ private ExecuteActivityParameters constructExecuteActivityParameters(
   }
 
   static RetryPolicy.Builder toRetryPolicy(RetryOptions retryOptions) {
+    String[] doNotRetry = retryOptions.getDoNotRetry();
+
     return RetryPolicy.newBuilder()
         .setInitialIntervalInSeconds(roundUpToSeconds(retryOptions.getInitialInterval()))
         .setMaximumIntervalInSeconds(roundUpToSeconds(retryOptions.getMaximumInterval()))
         .setBackoffCoefficient(retryOptions.getBackoffCoefficient())
         .setMaximumAttempts(retryOptions.getMaximumAttempts())
-        .addAllNonRetryableErrorTypes(Arrays.asList(retryOptions.getDoNotRetry()));
+        .addAllNonRetryableErrorTypes(
+            Arrays.asList(doNotRetry != null ? doNotRetry : new String[0]));
```

Comment:
[Nit] i can technically make this a private static variable that represents an empty array

Reply:
good point, i will omit that